bundles/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/handler/ModbusPollerThingHandler.java

```diff
@@ -246,21 +177,48 @@ public ModbusPollerReadRequest(ModbusPollerConfiguration config,
         }
     }
 
-    private final Logger logger = LoggerFactory.getLogger(ModbusPollerThingHandlerImpl.class);
+    /**
+     * Immutable data object to cache the results of a poll request
+     */
+    private class PollResult {
+        @Nullable
+        public final AsyncModbusReadResult result;
+        @Nullable
+        public final AsyncModbusFailure<ModbusReadRequestBlueprint> failure;
```

Comment:
```suggestion
        public final @Nullable AsyncModbusReadResult result;
        public final @Nullable AsyncModbusFailure<ModbusReadRequestBlueprint> failure;
```

Reply:
Resolved this, and many others in https://github.com/openhab/openhab-addons/pull/7994/commits/00b7f764be87d957e6b81354b54d7bdeb33cb818